### ISSUE TITLE
supervised learning on small datasets

### ISSUE THREAD

**Issue by nicolaspanel** [facebookresearch/fastText#241]
I'm currently working on a small dataset (~80k samples) using pretrained vectors from wikipedia.

To prevent overfitting, I would like to train the model without updating the word vectors (only the hidden layer). Is there a way to do that?

Best regards,
PS: thank you for this amazing tool!

**Comment by cpuhrsch** [comment]
Hello @nicolaspanel,

I assume you mean the input model parameters. There is currently no flag to prevent pretrained word vectors to not be updated. For that you'll need to keep track of the preloaded words and zero out their gradients during training. This is something you'd need to patch yourself for now. I'll add your feature request to our list of tasks. Sorry that I can't be more helpful than that at this time. I'm closing the task for now, but feel free to reopen if you don't think I've answered your question.

Thanks,
Christian

**Comment by matanster** [comment]
@nicolaspanel in case you patched this up as suggested, would you care to report back whether it had the intended noticeable effect on model performance, in this precise scenario?
